// filename: protos/audiostream.proto
syntax = "proto3";

// Defines the structure of a single chunk of audio data
message AudioChunk {
  bytes audio_chunk = 1;
}

// Defines the structure of the server's receipt message
message StreamReceipt {
  string status_message = 1;
}

// The gRPC service definition
service AudioStreamer {
  // A client-streaming RPC. The client sends a stream of AudioChunk messages.
  // The server responds with a single StreamReceipt when the client is done.
  rpc StreamAudio (stream AudioChunk) returns (StreamReceipt) {}
}